### DISCUSSION TITLE
Superset of MSBuild

### DISCUSSION THREAD

**Discussion by stan-sz** [dotnet/msbuild#9218] [Ideas]
I'm kicking off a discussion around MSBuild linting (https://github.com/dotnet/msbuild/issues/1777) - would it be beneficial to come up with a superset syntax for MSBuild, like Bicep for ARM and Type Script for Java Script, to allow more type-aware linting of MSBuild project files?

Ultimately I'm thinking of a VSCode-based developer experience where linting violations are signaled promptly without the need to kick off a build.

**Comment by rainersigwald** [comment]
Are you aware of https://github.com/mhutch/MonoDevelop.MSBuildEditor?

**Reply by stan-sz** [reply]
Thanks for the info. This is a preview VS-only extension. I have sparked the discussion to brainstorm a cross-plat, IDE-universal solution.

**Comment by GangWang01** [comment]
Close for long-time inactivity. Feel free to reopen if the answer doesn't satisfy you.
